src/Attribute.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1); // @codeCoverageIgnore
+
+namespace Rinq;
+
+/**
+ * Attribute is a session attribute.
+ *
+ * Sessions contain a versioned key/value store. See the Session interface for
+ * more information.
+ */
+final class Attribute
+{
+    /**
+     * Create a new attribute with the given key/value.
+     *
+     * @param string $key   The key is an application-defined identifier.
+     * @param mixed  $value The attributes value.
+     */
+    public static function create(string $key, mixed $value): self
+    {
+        return new self($key, $value, false);
+    }
+
+    /**
+     * Convenience method to create an attribute with the given key/value and
+     * marked as frozen.
+     *
+     * @param string $key   The key is an application-defined identifier.
+     * @param mixed  $value The attributes value.
+     */
+    public static function freeze(string $key, mixed $value): self
+    {
+        return new self($key, $value, true);
+    }
+
+    /**
+     * @return string The key is an application-defined identifier.
+     */
+    public function key(): string
+    {
+        return $this->key;
+    }
+
+    /**
+     * @return mixed  The attributes value.
+     */
+    public function value(): mixed
```

Comment:
I don't think you can typehint something as `mixed` - Not sure about PHP 7.2 but I'm pretty sure this would expect an actual class named `mixed`.

Still need to keep the `@return mixed` annotation, though.